tempesta_fw/t/functional/tests/test_frang.py

```diff
@@ -0,0 +1,516 @@
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+	def header_chunks(self):
+		global tcount
+
+		print("header_chunks\n")
+
+		part1 = b'GET / HTTP/1.0\r\n'
+		part2 = b'host: loc\r\n'
+		part3 = b'Connection: close\r\n\r\n'
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_header_chunk_cnt', '1')
+		self.cfg.add_end_of_section()
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(part1)
+		self.s.send(part2)
+		self.s.send(part3)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+		print("res:", self.res)
+
+	def header_timeout(self):
+		global tcount
+		part1 = b'GET / HTTP/1.0\r\n'
+		part2 = b'host: loc\r\n\r\n'
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('client_header_timeout', '1')
+		self.cfg.add_end_of_section()
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(part1)
+		time.sleep(1)
+		self.s.send(part2)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			tcount += 1
+			self.res = True
+			print( "res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+
+	def body_timeout(self):
+		global tcount
+		print("body_timeout")
+		part1 = b"POST /a.html HTTP/1.0\r\nHost: loc\r\n\
+Content-Type: text/html\r\nContent-Length: 20\r\n\r\n<html>content"
+		part2 = b"</html>\r\n\r\n"
+
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('client_body_timeout', '1')
+		self.cfg.add_end_of_section()
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(part1)
+		time.sleep(1)
+		self.s.send(part2)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+		print("res:", self.res)
+		time.sleep(5)
+		tfw.stop()
+
+	def body_chunks(self):
+		global tcount
+		print("body_chunks")
+		part1 = b"POST /a.html HTTP/1.0\r\nHost: loc\r\n\
+Content-Type: text/html\r\nContent-Length: 30\r\n\r\n<html><body>content"
+		part2 = b"</body>"
+		part3 = b"<html>\r\n\r\n"
+
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_ct_vals', '[\"text/html\"]')
+		self.cfg.add_option('http_body_chunk_cnt', '1')
+		self.cfg.add_end_of_section()
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(part1)
+		self.s.send(part2)
+		self.s.send(part3)
+		data = self.s.recv(1024)
+		if len(data) == 0:
```

Comment:
Does it mean that the connection was terminated? Can we satisfy the check if Tempestat is missbehaving and returns empty reposnse? I propose to grep dmesg (using some generic function of course, such that it can be used from many places) to check for frang limiting warning.